lib/recent.go

```diff
@@ -0,0 +1,107 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	OpenTofu  []string `json:"openTofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, distribution string) {
+	if !validVersionFormat(requestedVersion) {
+		logger.Errorf("The version %q is not a valid version string and won't be stored", requestedVersion)
+		return
+	}
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	if CheckFileExist(recentFilePath) {
+		unmarshalRecentFileData(recentFilePath, &recentFileData)
+	}
+	prependRecentVersionToList(requestedVersion, distribution, &recentFileData)
+	saveRecentFile(recentFileData, recentFilePath)
+}
+
+func prependRecentVersionToList(version, distribution string, r *RecentFiles) {
+	var sliceToCheck []string
+	if distribution == distributionTerraform {
+		sliceToCheck = r.Terraform
+	} else if distribution == distributionOpenTofu {
+		sliceToCheck = r.OpenTofu
+	}
+	for versionIndex, versionValue := range sliceToCheck {
+		if versionValue == version {
+			sliceToCheck = append(sliceToCheck[:versionIndex], sliceToCheck[versionIndex+1:]...)
+		}
+	}
+	sliceToCheck = append([]string{version}, sliceToCheck...)
+
+	if distribution == distributionTerraform {
+		r.Terraform = sliceToCheck
+	} else if distribution == distributionOpenTofu {
+		r.OpenTofu = sliceToCheck
+	}
+}
+
+func getRecentVersions(installPath string, dist string) ([]string, error) {
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	unmarshalRecentFileData(recentFilePath, &recentFileData)
+	var listOfRecentVersions []string
+	if dist == distributionTerraform {
+		listOfRecentVersions = recentFileData.Terraform
+	} else if dist == distributionOpenTofu {
+		listOfRecentVersions = recentFileData.OpenTofu
+	}
+	var maxCount int
+	if len(listOfRecentVersions) >= 3 {
+		maxCount = 3
```

Comment:
I might be mixing up things, though I seem to saw somewhere else(?) that @warrensbox said 5 would be better. Am I mixing different things?

Reply:
I just took over the existing value. If we want to change it, i missed it. @warrensbox

Reply:
I think it was mentioned in the other PR IIRC!

Reply:
Let's probably have @warrensbox confirm whether to go with the existing value or change it to 5 while we're here?

Reply:
https://github.com/warrensbox/terraform-switcher/pull/435/files#r1623089267
This is the comment where he saif to set it to 5.
Will do so